hist/hist/inc/TFormula.h

```diff
@@ -195,22 +203,43 @@ class TFormula : public TNamed
    Double_t       Eval(Double_t x, Double_t y , Double_t z , Double_t t ) const;
    Double_t       EvalPar(const Double_t *x, const Double_t *params=0) const;
 
+   /// Includes the clad header file if not already included
+   void IncludeCladRuntime();
+
    /// Generate gradient computation routine with respect to the parameters.
    /// \returns true if a gradient was generated and GradientPar can be called.
    bool GenerateGradientPar();
 
+   /// Generate hessian computation routine with respect to the parameters.
+   /// \returns true if a hessian was generated and HessianPar can be called.
+   bool GenerateHessianPar();
+
    /// Compute the gradient employing automatic differentiation.
    ///
    /// \param[in] x - The given variables, if nullptr the already stored
    ///                variables are used.
    /// \param[out] result - The result of the computation wrt each direction.
-   void GradientPar(const Double_t *x, TFormula::GradientStorage& result);
+   void GradientPar(const Double_t *x, TFormula::CladStorage& result);
 
    void GradientPar(const Double_t *x, Double_t *result);
 
+   /// Compute the gradient employing automatic differentiation.
+   ///
+   /// \param[in] x - The given variables, if nullptr the already stored
+   ///                variables are used.
+   /// \param[out] result - The result of the computation wrt each direction.
+   void HessianPar(const Double_t *x, TFormula::CladStorage& result);
+
```

Comment:
Here is missing some documentation for HessianPar. In particular how the Hessian matrix is stored in the return array.
Since the Hessian is a matrix, will not be better to have a signature with a matrix type or a std::vector<std::vector>> instead of an std::vector ?
Since the matrix is symmetric should we provide in the vector both upper and lower diagonal part ? Probably is more convenient (at the little cost of memory) to return both.

Reply:
`clad` produces the hessian matrix in a flattened form and de flattening it would have to occur on the ROOT side making the step very slow. So imo it's better to give user the flattened matrix rather than giving them a slower function.

Reply:
Thanks for adding some documentation. 
I would also add that the matrix is flattens to form a vector in row-major order